src/intersection-observer-polyfill.js

```diff
@@ -376,22 +381,37 @@ export class IntersectionObserverPolyfill {
    * @param {!ElementIntersectionStateDef} state
    * @param {!./layout-rect.LayoutRectDef} hostViewport hostViewport's rect
    * @param {./layout-rect.LayoutRectDef=} opt_iframe iframe container rect
+   *    If opt_iframe is provided, all LayoutRect has position relative to
+   *    the iframe. If opt_iframe is not provided,
+   *    all LayoutRect has position relative to the host document.
+   * @param {boolean=} opt_calledFromMutationObserver
+   *    Whether this was called from a mutation observer
+   *    This will signal that the layout of the elements
+   *    must use getBoundingClientRect, as Resource Scheduler 
+   *    could be called too late, and not clear the layout cache.
    * @return {?IntersectionObserverEntry} A valid change entry or null if ratio
    * @private
    */
-  getValidIntersectionChangeEntry_(state, hostViewport, opt_iframe) {
+  getValidIntersectionChangeEntry_(state, hostViewport, 
+    opt_iframe, opt_calledFromMutationObserver) {
     const {element} = state;
 
     // Normalize container LayoutRect to be relative to page
+    let elementRect = null;
+    const owner = element.getOwner();
     let ownerRect = null;
 
-    // If opt_iframe is provided, all LayoutRect has position relative to
-    // the iframe.
-    // If opt_iframe is not provided, all LayoutRect has position relative to
-    // the host document.
-    const elementRect = element.getLayoutBox();
-    const owner = element.getOwner();
-    ownerRect = owner && owner.getLayoutBox();
+    // If opt_calledFromMutationObserver is provided, 
+    // we need to get our bounding rect rather
+    // Than our layout box, in order to avoid race conditions 
+    // with the resource scheduler
+    if (opt_calledFromMutationObserver) {
+      elementRect = element./*OK*/getBoundingClientRect();
```

Comment:
@torch2424 `elementRect` is the rect relative to the document not viewport. We can't use `getBoundingClientRect()` directly.